bindings/cs/Scripts/build.cake

```diff
@@ -0,0 +1,214 @@
+var releaseVersion = GetSettingValue("releaseVersion", localReleaseVersion);
+var fork = GetSettingValue("fork", "https://github.com/mongodb/libmongocrypt.git");
+var branch = GetSettingValue("branch", "master"); 
+var libmongocryptAllUrl = GetSettingValue("url", "https://mciuploads.s3.amazonaws.com/libmongocrypt/all/master/508e21f4abff9f5519e0357a63a4ad30d2c24692/libmongocrypt-all.tar.gz");
+var csharpBindingsGitTagName = $"csharp-v{releaseVersion}";
+var csharpBindingsDirectory = buildDirectory.Combine(csharpBindingsGitTagName);
+var libmongocryptRelWithDebInfoDirectory = csharpBindingsDirectory.Combine("cmake-build").Combine("RelWithDebInfo");
+var libmongocryptCsDirectory = csharpBindingsDirectory.Combine("bindings").Combine("cs");
+var libmongocryptSolutionDirectory = libmongocryptCsDirectory.Combine("MongoDB.Libmongocrypt");
+var libmongocryptSolutionFile = libmongocryptSolutionDirectory.CombineWithFilePath("MongoDB.Libmongocrypt.csproj");
+var libmongocryptTestsSolutionDirectory = libmongocryptCsDirectory.Combine("MongoDB.Libmongocrypt.Test");
+var artifactsDirectory = buildDirectory.Combine("artifacts");
+
+Task("Prepare")
+    .Does(() =>
+    {
+        if (DirectoryExists(buildDirectory))
+        {
+            DeleteDirectory(
+                buildDirectory, 
+                new DeleteDirectorySettings {
+                    Recursive = true,
+                    Force = true
+                });
+        }
+        CreateDirectory(buildDirectory);
+
+        Information($"Release version: {releaseVersion}");
+        Information($"Fork: {fork}");
+        Information($"Branch: {branch}");
+        Information($"Native libraries url: {libmongocryptAllUrl}");
+
+        Information("Downloading native libs..");
+        EnsureDirectoryExists(libmongocryptAllDirectory);
+        var nativeLibrariesArchive = libmongocryptAllDirectory.CombineWithFilePath("libmongocrypt-all.tar");
+        DownloadFile(libmongocryptAllUrl, nativeLibrariesArchive);
+        
+        Information("Unzipping..");
+        UncompressToTheCurrentDirectory(nativeLibrariesArchive);
+
+        Information("Cloning the libmongocrypt repo..");
+        GitClone(
+            fork, 
+            csharpBindingsDirectory, 
+            new GitCloneSettings
+            {
+                BranchName = branch,
+                Checkout = true,
+                IsBare = false,
+                RecurseSubmodules = true
+            });
+
+        EnsureDirectoryExists(libmongocryptRelWithDebInfoDirectory);
+        EnsureDirectoryExists(downloadedMongocryptDirectory);
+        CopyFile(
+            libmongocryptAllDirectory.Combine("windows-test").Combine("bin").CombineWithFilePath("mongocrypt.dll"),
+            downloadedMongocryptDirectory.CombineWithFilePath("mongocrypt.dll"));
+        CopyFile(
+            libmongocryptAllDirectory.Combine("ubuntu1804-64").Combine("nocrypto").Combine("lib").CombineWithFilePath("libmongocrypt.so"),
```

Comment:
What are these `nocrypto` directories? It sounds like versions of the binaries that expose the desired API but do not perform any crypto. Shouldn't we be packaging up the same binaries but without `nocrypto` in the pathname?

Reply:
nope, this related to this ticket (and macos one): https://jira.mongodb.org/browse/MONGOCRYPT-86. In two words, in Linux and MacOS, we configure crypto callbacks like here https://github.com/mongodb/libmongocrypt/blob/master/bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs#L49, so the C level calls these hooks to make signing/hashing/encryption and so on steps. But in windows, this stuff is built-in to the C library

Reply:
Just wanted to make sure we were packing the right binaries. You're much more familiar with the details. So I'll trust you on this.